waku/waku_rln_relay/rln_relay.nim

```
@@ -319,6 +331,9 @@ proc appendRLNProof*(
   ## `senderEpochTime` indicates the number of seconds passed since Unix epoch. The fractional part holds sub-seconds.
   ## The `epoch` field of `RateLimitProof` is derived from the provided `senderEpochTime` (using `calcEpoch()`)
 
+  # need to update the timestamp to the sender's timestamp bcz now we are using the sender's timestamp to generate the proof
+  msg.timestamp = senderEpochTime.int64
```

I am not sure I understand why this is necessary. Shouldn't the `msg.timestamp` here be the sender time in any case? Under what circumstance would `senderEpochTime` be computed on a different timestamp? What does change is that `timestamp` is now a compulsory field and we can return an Error if the message does not include a timestamp.

---

Thanks ! There are some test cases that initialize WakuMessage and generate a proof at different times. Initially, I handled that logic here, but I now realize that approach was overfitting. Instead of modifying it here, I’ve updated the relevant test cases instead.